Kernel/Interrupts/IRQController.h

```diff
@@ -36,7 +37,7 @@ enum class IRQControllerType {
     i82093AA = 2 /* Intel 82093AA I/O ADVANCED PROGRAMMABLE INTERRUPT CONTROLLER (IOAPIC) */
 };
 
-class IRQController {
+class IRQController : public RefCounted<IRQController> {
```

Comment:
When we make something ref-counted, we must make sure to `adopt` the initial ref wherever we do `new` (or it is guaranteed to leak.)

```cpp
auto controller = adopt(*new IRQController);
```